### PR TITLE
Some training improvements

### PR DESCRIPTION
**Pull request by CiaraStrawberry**
- Added gradient accumulation functionality
- Fixed issue where this generates a state dict that is incompatible with running this on non distributed torch code
- Automatically checkpoints only the motion module during training.

### CODE REVIEW

**Comment by CiaraStrawberry** [comment]
whoops updated the main branch,
